runtime/lib/ttnn/operations/pool/maxpool2d.cpp

```diff
@@ -56,6 +56,10 @@ void run(const ::tt::target::ttnn::MaxPool2dOp *op, ProgramContext &context) {
                        {op->padding_height(), op->padding_width()},
                        {op->dilation_height(), op->dilation_width()}, &device);
 
+  auto new_memconfig = out.memory_config();
+  new_memconfig.memory_layout = TensorMemoryLayout::INTERLEAVED;
+  new_memconfig.buffer_type = BufferType::DRAM;
+  out = ::ttnn::to_memory_config(out, new_memconfig, std::nullopt);
```

Comment:
This needs a block surrounding it with workaround comment + issue number.

Reply:
Can we add some FS/override for this so at least it can be turned off when needed?

Reply:
@jnie-TT have you given any more thought to the override environment thing we spoke about on slack?  I think this PR is a perfect use case for a feature like that.

Reply:
@nsmithtt yeah this seems like a great application for it. I think going with the `wa::Env` approach makes sense, every override flag should have a ticket linked to it, the change that solves the ticket should include removing the workaround from `wa::Env`.

@LPanosTT for context, we were thinking of adding a workaround namespace in runtime alongside the `debug::Env`. The workaround namespace contains a shared Env that contains boolean flags that signal whether or not to enable a workaround. Every workaround should be wrapped with an if statement checking if we want to enable the workaround. This way:

- Workarounds don't go out of hand.
- Workarounds can be tracked easily. Once one is solved, the workaround should be disabled everywhere i.e. the change that solves the ticket should include removing the workaround from `wa::Env`. 
- We have a flip-switch method of enabling/disabling certain workarounds - just need to set the flag to true/false.
- Centralized control where we track all workarounds in one place, so a user can know directly what hacks are being made in one glance, and it's clear which tickets correspond to which workarounds. 

Lmk what you think @LPanosTT !